Camera2SlowMotion/app/src/main/java/com/example/android/camera2/slowmo/CameraActivity.kt

```diff
@@ -17,18 +17,41 @@
 package com.example.android.camera2.slowmo
 
 import android.os.Bundle
+import android.util.Log
 import android.view.View
 import android.widget.FrameLayout
 import androidx.appcompat.app.AppCompatActivity
+import java.io.BufferedReader
+import java.io.IOException
+import java.io.InputStreamReader
+import java.io.PrintWriter
+import java.net.Socket
 
 class CameraActivity : AppCompatActivity() {
 
     private lateinit var container: FrameLayout
+    private var SERVER_IP = "127.0.0.1"
```

Comment:
This looks a little strange as it is actually the IP of the phone.  I assume this is because you are port forwarding, but maybe the SERVER_IP name is not the best.  Its ok to leaveas I don't really have a better suggestion.

Reply:
This was done on purpose as the `SERVER_IP` and be changed to a network IP too, and the same implementation will work. Here currenlty I'm using a localhost becasue of port-forwarding.

Reply:
I think you will need to change a lot more than just a hard-coded IP address if you change to connecting to the server by IP.

Also, this should be "const val" rather than var if you are treating it as a constant (CAPITAL_CASE)